src/flamenco/runtime/program/fd_system_program_nonce.c

```diff
@@ -928,95 +929,118 @@ fd_load_nonce_account( fd_exec_txn_ctx_t const *   txn_ctx,
   fd_txn_instr_t const * txn_instr = &txn_ctx->txn_descriptor->instr[0];
   fd_acct_addr_t const * tx_accs = fd_txn_get_acct_addrs( txn_ctx->txn_descriptor, txn_ctx->_txn_raw->raw );
   fd_acct_addr_t const * prog_id = tx_accs + txn_instr->program_id;
-
-  if( memcmp( prog_id->b, fd_solana_system_program_id.key, sizeof( fd_pubkey_t ) ) ) {
-    return 0;
+  if( FD_UNLIKELY( memcmp( prog_id->b, fd_solana_system_program_id.key, sizeof( fd_pubkey_t ) ) ) ) {
+    return FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
   }
-
   uchar const * instr_data  = fd_txn_get_instr_data( txn_instr, txn_ctx->_txn_raw->raw );
   uchar const * instr_accts = fd_txn_get_instr_accts( txn_instr, txn_ctx->_txn_raw->raw );
 
-  if( FD_UNLIKELY( txn_instr->data_sz != 4UL || FD_LOAD( uint, instr_data ) != 
-                    (uint)fd_system_program_instruction_enum_advance_nonce_account ) ) {
-    return 0;
+  if( FD_UNLIKELY( txn_instr->data_sz!=4UL || FD_LOAD( uint, instr_data ) != 
+                   (uint)fd_system_program_instruction_enum_advance_nonce_account ) ) {
+    return FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
   }
-
-  const fd_pubkey_t * durable_nonce_acc = &txn_ctx->accounts[ instr_accts[0] ];
-  
-  if( fd_txn_is_writable( txn_ctx->txn_descriptor, instr_accts[0] ) == 0 ) {
-    return 0;
+  if( FD_UNLIKELY( !fd_txn_is_writable( txn_ctx->txn_descriptor, instr_accts[0] ) ) ) {
+    return FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
   }
 
   FD_BORROWED_ACCOUNT_DECL( durable_nonce_rec );
-  int err = fd_acc_mgr_view( txn_ctx->acc_mgr, txn_ctx->slot_ctx->funk_txn, (fd_pubkey_t const *)durable_nonce_acc, durable_nonce_rec );
-
-  if( FD_UNLIKELY( err != FD_ACC_MGR_SUCCESS ) ) {
-    *perr = FD_EXECUTOR_INSTR_ERR_CUSTOM_ERR;
-    return 0;
+  int err = fd_acc_mgr_view( txn_ctx->acc_mgr, txn_ctx->slot_ctx->funk_txn, &txn_ctx->accounts[ instr_accts[0] ], durable_nonce_rec );
+  if( FD_UNLIKELY( err!=FD_ACC_MGR_SUCCESS ) ) {
+    return FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
   }
 
-  /* https://github.com/firedancer-io/solana/blob/4b31032e68f85848b02fcc4c9e580d57f32ec04b/sdk/src/nonce_account.rs#L30 */
+  /* https://github.com/anza-xyz/agave/blob/16de8b75ebcd57022409b422de557dd37b1de8db/sdk/src/nonce_account.rs#L28-L42 */
+  /* verify_nonce_account */  
   uchar const * owner_pubkey = durable_nonce_rec->const_meta->info.owner;
-  if( memcmp( owner_pubkey, fd_solana_system_program_id.key, sizeof( fd_pubkey_t ) ) ) {
-    return 0;
+  if( FD_UNLIKELY( memcmp( owner_pubkey, fd_solana_system_program_id.key, sizeof( fd_pubkey_t ) ) ) ) {
+    return FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
   }
 
-  /* https://github.com/firedancer-io/solana/blob/4b31032e68f85848b02fcc4c9e580d57f32ec04b/sdk/program/src/nonce/state/mod.rs#L38 */
   fd_bincode_decode_ctx_t decode = {
     .data    = durable_nonce_rec->const_data,
     .dataend = durable_nonce_rec->const_data + durable_nonce_rec->const_meta->dlen,
-    .valloc  = fd_scratch_virtual()
+    .valloc  = txn_ctx->valloc
```

Comment:
What is the reasoning here to not use scratch?

Reply:
this was a bad change, reverting